### PR TITLE
[fix][transaction] Handle some failures in transaction

### PR DESCRIPTION
**Pull request by gaoran10**
### Modifications

Handle some failures while writing transaction logs and markers.

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/ProducerStateManager.java [hunk 1]
```diff
@@ -178,7 +178,7 @@ void updateAbortedTxnsPurgeOffset(long abortedTxnsPurgeOffset) {
     }
 
     long maybePurgeAbortedTx() {
-        if (mapEndOffset == -1) {
+        if (mapEndOffset == -1 || kafkaTxnPurgeAbortedTxnIntervalSeconds <= 0) {
             return 0;
         }
         long now = System.currentTimeMillis();
```
Comment (Demogorgon314) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/ProducerStateManager.java:184]: I find the `deltaFromLast` has been divided by 1000 twice. We should only divide it once.

Comment (gaoran10) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/ProducerStateManager.java:184]: It seems that this issue had been fixed.

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerChannelManager.java [hunk 2]
```diff
@@ -200,8 +201,21 @@ public CompletableFuture<TransactionMarkerChannelHandler> getChannel(InetSocketA
                         handlerFuture.completeExceptionally(e);
                         return null;
                     });
+
+            // do not cache failed futures
+            handlerFuture.exceptionally(error -> {
+                log.error("getChannel failed {} {}", socketAddress, error.getMessage(), error);
+                handlerMap.remove(socketAddress, handlerFuture);
```
Comment (Demogorgon314) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerChannelManager.java:208]: Removing the value in the map may cause `Recursive update` error.

**Review by Demogorgon314 — CHANGES_REQUESTED** [review]

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerChannelManager.java (hunk 2 above)
Comment (Demogorgon314) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerChannelManager.java:208]: Here is an example:
```java
ConcurrentHashMap<String, String> map = new ConcurrentHashMap<>();
        map.computeIfAbsent("test", key -> {
            map.remove("test");
            return "test";
        });
```

**Pull request merged**